pkg/client/kubelet.go

```diff
@@ -135,7 +136,43 @@ func (c *HTTPKubeletClient) GetPodStatus(host, podNamespace, podID string) (api.
 }
 
 func (c *HTTPKubeletClient) HealthCheck(host string) (probe.Result, error) {
-	return httprobe.DoHTTPProbe(fmt.Sprintf("%s/healthz", c.url(host)), c.Client)
```

Comment:
Hmm... The issue is the discrepancy in the hostname between the kubelet and the master. So you need to compare information from both sides to catch it.